src/core/api/replication_policy.go

```diff
@@ -128,6 +128,18 @@ func (pa *RepPolicyAPI) Post() {
 	policy := &api_models.ReplicationPolicy{}
 	pa.DecodeJSONReqAndValidate(policy)
 
+	// check the name
+	exist, err := exist(policy.Name)
+	if err != nil {
+		pa.HandleInternalServerError(fmt.Sprintf("failed to check the existence of policy %s: %v", policy.Name, err))
+		return
+	}
+
+	if exist {
+		pa.HandleConflict("name is already used")
```

Comment:
pa.HandleConflict(fmt.Sprintf("the policy name %s already exists", policy.Name))